docs/configure-artifact-repository.md

```diff
@@ -107,43 +110,176 @@ $ cat > policy.json <<EOF
+
+```bash
+aws iam create-role --role-name $mybucket-role
+aws iam put-role-policy --role-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
+```
+
+To use S3 IRSA, you need to annotate the service account of both the running workflow (in order to save logs/artifacts) and the argo-server pod (in order to retrieve the logs/artifacts).
+You can attach only one IAM role to a service account, so your roles must have permission for all buckets you may wish to use.
 
 ```yaml
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   annotations:
-    eks.amazonaws.com/role-arn: arn:aws:iam::012345678901:role/mybucket
+    eks.amazonaws.com/role-arn: arn:aws:iam::012345678901:role/mybucket-role
   name: myserviceaccount
   namespace: mynamespace
 ```
 
+You can then use the service account in a workflow as below:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+spec:
+  serviceAccountName: myserviceaccount
+```
+
+### AWS S3 with IAM user credentials
+
+AWS access keys have the same permissions as the user they are associated with.
+In particular, you cannot create access keys with reduced scope.
+If you want to limit the permissions for an access key, you will need to create a user with just the permissions you want to associate with the access key.
+Otherwise, you can just create an access key using your existing user account.
+
+If you have created a bucket and policy as described above, you can create an IAM user, add the policy and create a set of access keys with:
+
+```bash
+aws iam create-user --user-name $mybucket-user
+aws iam put-user-policy --user-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
+aws iam create-access-key --user-name $mybucket-user > access-key.json
+```
+
+You can then use the access keys by configuring your artifact as below:
+
+```yaml
+artifacts:
+  - name: my-output-artifact
+    path: /my-output-artifact
+    s3:
+      endpoint: s3.amazonaws.com
+      bucket: my-s3-bucket
+      key: path/in/bucket/my-output-artifact.tgz
+      # The following secret entries are secret selectors.
+      # They reference the k8s secret named 'my-s3-credentials'.
+      # This secret is expected to have the keys 'accessKey' and 'secretKey',
+      # containing the base64 encoded credentials to the bucket.
+      accessKeySecret:
+        name: my-s3-credentials
+        key: accessKey
+      secretKeySecret:
+        name: my-s3-credentials
+        key: secretKey
+```
+
+### AWS S3 with IAM role credentials
+
+You can use IAM role credentials for temporary access.
+
+If you have created a bucket and policy as described above, you can then create an IAM role and add the policy with:
+
+```bash
+aws iam create-role --role-name $mybucket-role
+aws iam put-role-policy --role-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
+```
+
+You can then obtain the credentials set by running:
```

Comment:
```suggestion
Retrieve the role credentials:
```

- simplify + active voice
- also used "retrieve" instead of "obtain" as a more common word